src/Microsoft.ML.OnnxConverter/OnnxExportExtensions.cs

```diff
@@ -26,13 +37,23 @@ internal static ModelProto ConvertToOnnxProtobuf(this ModelOperationsCatalog cat
         {
             var env = catalog.GetEnvironment();
             var ctx = new OnnxContextImpl(env, "model", "ML.NET", "0", 0, "machinelearning.dotnet", OnnxVersion.Stable);
-            var outputData = transform.Transform(inputData);
-            LinkedList<ITransformCanSaveOnnx> transforms = null;
-            using (var ch = env.Start("ONNX conversion"))
-            {
-                SaveOnnxCommand.GetPipe(ctx, ch, outputData, out IDataView root, out IDataView sink, out transforms);
-                return SaveOnnxCommand.ConvertTransformListToOnnxModel(ctx, ch, root, sink, transforms, null, null);
-            }
+            return ConvertToOnnxProtobufCore(env, ctx, transform, inputData);
+        }
+
+        /// <summary>
+        /// Convert the specified <see cref="ITransformer"/> to ONNX format. Note that ONNX uses Google's Protobuf so the returned value is a Protobuf object.
+        /// </summary>
+        /// <param name="catalog">The class that <see cref="ConvertToOnnxProtobuf(ModelOperationsCatalog, ITransformer, IDataView, int)"/> attached to.</param>
+        /// <param name="transform">The <see cref="ITransformer"/> that will be converted into ONNX format.</param>
+        /// <param name="inputData">The input of the specified transform.</param>
+        /// <param name="opSetVersion">Set custom value to OpSet version</param>
```

Comment:
Would it be sensible to add a note saying that currently we only support versions higher than 9, and perhaps that each component has a different minimum version?

Reply:
makes sense, added